### PR TITLE
Search bar

### PR DESCRIPTION
**Pull request by DustCreech**

### CODE REVIEW

**Review by aasch2020 — COMMENTED** [review]
I see a buncha files for Nodejs, was that added to the project?

File: controllers.py [hunk 1]
```diff
@@ -33,22 +33,28 @@
...
 url_signer = URLSigner(session)
 
 
 @action('index')
 @action.uses('index.html', db)
 def index():
-
-    return dict()
+    # Section is for the searchBar component
+    userinput = request.params.get('userinput')
```
Comment (jonner16) [inline controllers.py]: Following standard python formatting this should be user_input

File: controllers.py [hunk 2]
```diff
@@ -33,22 +33,28 @@
...
+    if userinput == "" or userinput is None:
+        results = db(db.observations_na).select(limitby=(0,10))
+    else:
+        results = db(db.observations_na.species_guess.contains(userinput, all=True)).select(limitby=(0,10))
```
Comment (jonner16) [inline controllers.py]: I wouldnt search solely off of species_guess as that is user inputted, so perhaps search off of scientific_name and ideally search off of multiple params: species_guess,scientific_name,common_name,iconic_taxon_name

**Review by jonner16 — CHANGES_REQUESTED** [review]
Overall it looks good! Just a few changes needed before getting merged.

**Review by jonner16 — APPROVED** [review]
Tested search with multiple params and it works great! Good job

**Review by jonner16 — APPROVED** [review]
Tested search with multiple params and it works great! Good job

**Review by aasch2020 — APPROVED** [review]

**Review by MTGrules — APPROVED** [review]

**Pull request merged**